include/aspect/postprocess/visualization/strain_rate_tensor_dike_correction.h

```diff
@@ -0,0 +1,66 @@
+/*
+  Copyright (C) 2015 - 2021 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef _aspect_postprocess_visualization_strain_rate_tensor_dike_correction_h
+#define _aspect_postprocess_visualization_strain_rate_tensor_dike_correction_h
+
+#include <aspect/postprocess/visualization.h>
+#include <aspect/simulator_access.h>
+
+#include <deal.II/numerics/data_postprocessor.h>
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      /**
+       * A class derived from DataPostprocessor that takes an output vector
+       * and computes a variable that represents the 4 or 9
+       * components (in 2d and 3d, respectively) of the strain rate tensor every
+       * point, which is corrected for implementing the dike injection process.
+       * 
+       * The strain rate is defined as $\varepsilon(\mathbf u)$ in the incompressible
+       * case and as $\varepsilon(\mathbf u)
+       * - \tfrac 13 (\textrm{trace}\ \varepsilon(\mathbf u)) \mathbf I$
+       * in the compressible case.
+       *
+       * The member functions are all implementations of those declared in the
+       * base class. See there for their meaning.
+       */
+      template <int dim>
+      class DikeStrainRateTensor
```

Comment:
```suggestion
      class StrainRateTensorDikeCorrection
```
or, even better,
```suggestion
      class StrainRateTensorDikingCorrected
```
That would require changing the file names etc accordingly though.